clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -650,7 +650,10 @@ class DiagnoseHLSLAvailability
   bool HasMatchingEnvironmentOrNone(const AvailabilityAttr *AA);
 
 public:
-  DiagnoseHLSLAvailability(Sema &SemaRef) : SemaRef(SemaRef) {}
+  DiagnoseHLSLAvailability(Sema &SemaRef)
+      : SemaRef(SemaRef),
+        CurrentShaderEnvironment(llvm::Triple::UnknownEnvironment),
```

Comment:
Environment could be gotten from SemaRef.getASTContext().getTargetInfo().getTriple().getEnvironment().

Reply:
Right, but this is just the default initialization, these values are not used. They get updated in `SetShaderStageContext` or `SetShaderStageContext` before any analysis is done. This environment value will also be changing when for libraries libraries depending on which entry point is getting scanned.